src/git-ship

```diff
@@ -2,6 +2,19 @@
 source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
 
 
+function ensure_has_no_unshipped_ancestors {
+  local branch_name=$1
+
+  local parent_branch_name ; parent_branch_name=$(parent_branch "$branch_name")
+  if [ "$parent_branch_name" != "$MAIN_BRANCH_NAME" ]; then
+    echo_error_header
+    echo_error "Shipping this branch would ship '$parent_branch_name' as well."
```

Comment:
Not a bad idea. This is a larger change, though, that requires a good amount of extra infrastructure for list formatting. Let's do this separately in #562 to keep this PR focussed. The current behavior is good enough for the being, and not wrong per se.